src/Concordium/Client/Output.hs

```diff
@@ -1096,7 +1101,37 @@ showRejectReason verbose = \case
     Types.NonExistentTokenId tokenId -> printf "token id %s does not exist on-chain" (show tokenId)
     Types.TokenHolderTransactionFailed reason -> do
         let details = Types.tmrrDetails reason
-            invalidCBOR = printf "token holder transaction failed: %s (Could not decode reject details as valid CBOR)\n" (show reason)
+            invalidCBOR =
+                printf
+                    "%s token-holder transaction was rejected due to: %s; details (undecoded CBOR): %s"
```

Comment:
I'd suggest putting the details (whether decoded or not) on a new line each time.